lmnet/tests/lmnet_tests/datasets_tests/test_div2k.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import os
+from glob import glob
+
+import numpy as np
+import pytest
+
+from lmnet import environment
+from lmnet.datasets.dataset_iterator import DatasetIterator
+from lmnet.datasets.div2k import Div2k
+
+
+def test_train_files(set_test_environment):
+    dataset = Div2k("train")
+    files = sorted(dataset.files)
+
+    expected = [
+        "tests/fixtures/datasets/DIV2K/DIV2K_train_HR/0001.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_train_HR/0002.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_train_HR/0003.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_train_HR/0004.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_train_HR/0005.png",
+    ]
+
+    assert files == expected
+
+
+def test_validation_files(set_test_environment):
+    dataset = Div2k("validation")
+    files = sorted(dataset.files)
+
+    expected = [
+        "tests/fixtures/datasets/DIV2K/DIV2K_valid_HR/0001.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_valid_HR/0002.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_valid_HR/0003.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_valid_HR/0004.png",
+        "tests/fixtures/datasets/DIV2K/DIV2K_valid_HR/0005.png",
+    ]
+
+    assert files == expected
+
+
+def test_length(set_test_environment):
+    dataset = Div2k("train")
+    expected = 5
+
+    assert len(dataset) == expected
+
+
+def test_num_per_epoch(set_test_environment):
+    dataset = Div2k("train")
+    expected = 5
+
+    assert dataset.num_per_epoch == expected
+
+
+def test_get_item(set_test_environment):
+    dataset = Div2k("train")
+
+    assert all([isinstance(image, np.ndarray) for image, _ in dataset])
```

Comment:
no need to use list (`[]`) here. Generator comprehension is better as it'll fail early.

Reply:
I did not know that!
thank you!!